test/infrastructure/docker/api/v1beta1/dockermachinetemplate_webhook_test.go

```diff
@@ -51,10 +60,22 @@ func TestDockerMachineTemplateInvalid(t *testing.T) {
 			oldTemplate: &oldTemplate,
 			wantError:   true,
 		},
+		{
+			name:        "skip immutability check",
```

Comment:
nit: Should we also add test case for only dry-run and only the annotation being set?